pennylane_forest/qc.py

```diff
@@ -0,0 +1,296 @@
+            for qubits to decay to the ground state naturally.
+            Setting this to ``True`` results in a significantly faster expectation value
+            evaluation when the number of shots is larger than ~1000.
+
+    Keyword args:
+        compiler_timeout (int): number of seconds to wait for a response from quilc (default 10).
+        execution_timeout (int): number of seconds to wait for a response from the QVM (default 10).
+        parametric_compilation (bool): a boolean value of whether or not to use parametric
+            compilation.
+    """
+    pennylane_requires = ">=0.17"
+    version = __version__
+    author = "Rigetti Computing Inc."
+
+    def __init__(self, device, *, shots=1000, wires=None, active_reset=False, **kwargs):
+        if shots is not None and shots <= 0:
+            raise ValueError("Number of shots must be a positive integer or None.")
+
+        self._compiled_program = None
+
+        self.parametric_compilation = kwargs.get("parametric_compilation", True)
+
+        if self.parametric_compilation:
+            self._circuit_hash = None
+            """None or int: stores the hash of the circuit from the last execution which
+            can be used for parametric compilation."""
+
+            self._compiled_program_dict = {}
+            """dict[int, pyquil.ExecutableDesignator]: stores circuit hashes associated
+                with the corresponding compiled programs."""
+
+            self._parameter_map = {}
+            """dict[str, float]: stores the string of symbolic parameters associated with
+                their numeric values. This map will be used to bind parameters in a parametric
+                program using PyQuil."""
+
+            self._parameter_reference_map = {}
+            """dict[str, pyquil.quilatom.MemoryReference]: stores the string of symbolic
+                parameters associated with their PyQuil memory references."""
+
+        timeout_args = self._get_timeout_args(**kwargs)
+
+        self.qc = self.get_qc(device, **timeout_args)
+
+        self.num_wires = len(self.qc.qubits())
+
+        if wires is None:
+            # infer the number of modes from the device specs
+            # and use consecutive integer wire labels
+            wires = range(self.num_wires)
+
+        if isinstance(wires, int):
+            raise ValueError(
+                f"Device has a fixed number of {self.num_wires} qubits. The wires argument can only be used "
+                "to specify an iterable of wire labels."
+            )
+
+        if self.num_wires != len(wires):
+            raise ValueError(
+                f"Device has a fixed number of {self.num_wires} qubits and "
+                f"cannot be created with {len(wires)} wires."
+            )
+
+        self.wiring = dict(enumerate(self.qc.qubits()))
+        self.active_reset = active_reset
+
+        super().__init__(wires, shots)
+        self.reset()
```

Comment:
`self.reset()` is already called in `super().__init__`

Reply:
Good catch, removed it.